pkg/migrations/op_create_table_test.go

```diff
@@ -76,6 +76,72 @@ func TestCreateTable(t *testing.T) {
 				}, rows)
 			},
 		},
+		{
+			name: "create table with composite key",
+			migrations: []migrations.Migration{
+				{
+					Name: "01_create_table",
+					Operations: migrations.Operations{
+						&migrations.OpCreateTable{
+							Name: "users",
+							Columns: []migrations.Column{
+								{
+									Name: "id",
+									Type: "serial",
+									Pk:   ptr(true),
+								},
+								{
+									Name: "rand",
+									Type: "varchar(255)",
+									Pk:   ptr(true),
+								},
+								{
+									Name:   "name",
+									Type:   "varchar(255)",
+									Unique: ptr(true),
+								},
+							},
+						},
+					},
+				},
+			},
+			afterStart: func(t *testing.T, db *sql.DB, schema string) {
+				// The new view exists in the new version schema.
+				ViewMustExist(t, db, schema, "01_create_table", "users")
+
+				// Data can be inserted into the new view.
+				MustInsert(t, db, schema, "01_create_table", "users", map[string]string{
+					"rand": "123",
+					"name": "Alice",
+				})
+
+				// Data can be retrieved from the new view.
+				rows := MustSelect(t, db, schema, "01_create_table", "users")
+				assert.Equal(t, []map[string]any{
+					{"id": 1, "rand": "123", "name": "Alice"},
+				}, rows)
+			},
+			afterRollback: func(t *testing.T, db *sql.DB, schema string) {
+				// The underlying table has been dropped.
+				TableMustNotExist(t, db, schema, "users")
+			},
+			afterComplete: func(t *testing.T, db *sql.DB, schema string) {
+				// The view still exists
+				ViewMustExist(t, db, schema, "01_create_table", "users")
+
+				// Data can be inserted into the new view.
+				MustInsert(t, db, schema, "01_create_table", "users", map[string]string{
+					"rand": "123",
+					"name": "Alice",
+				})
+
+				// Data can be retrieved from the new view.
+				rows := MustSelect(t, db, schema, "01_create_table", "users")
+				assert.Equal(t, []map[string]any{
+					{"id": 1, "rand": "123", "name": "Alice"},
+				}, rows)
```

Comment:
I think what we want to assert in this test is that the columns that were marked as a`pk: true` in the migration really are part of the table's primary key.

We can get this information from the system catalogs and have a new assertion like `ColumnMustBePK` in `op_common_test.go`.

Alternatively, the tests could try to insert a duplicate value and assert that the insertion fails with a unique constraint error using the `MustNotInsert` assertion.

Reply:
Updated test.